src/carousel/carousel.controller.ts

```diff
@@ -105,21 +139,31 @@ export class CarouselController {
   @Patch('slides/update/:id/image')
   async updateImage(
     @Param('id') id: string,
+    @Query() query: LanguageDto,
     @UploadedFile() image: Express.Multer.File,
   ): Promise<CarouselSlideDto> {
+    const language = query.lang ?? Language.En;
     const updatedCarouselSlide = await this.carouselService.updateFile(
       +id,
       image,
       'image',
+      language,
     );
     return toCarouselSlideDto(updatedCarouselSlide);
   }
 
   /* Make carousel slide expire */
   @Roles(Role.Superadmin)
   @Patch('slides/expire/:id')
-  async expire(@Param('id') id: string): Promise<CarouselSlideDto> {
-    const expiredCarouselSlide = await this.carouselService.expire(+id);
+  async expire(
+    @Param('id') id: string,
+    @Query() query: LanguageDto,
+  ): Promise<CarouselSlideDto> {
+    const language = query.lang ?? Language.En;
+    const expiredCarouselSlide = await this.carouselService.expire(
+      +id,
+      language,
+    );
```

Comment:
The expire function should be language agnostic. The logic is as follows: sometimes the admin might want to deprecate the slide because it's outdated, and we don't want to delete any slides (as to not lose important assets from the past).

It simply sets the `expiresAt` to `Date.now()` which is language agnostic.

Reply:
so maybe we can return nothing in expire function, since returning we have to specify a language